tests/hazmat/backends/test_openssl.py

```diff
@@ -282,6 +269,14 @@ def test_activate_builtin_random_already_active(self):
         e = backend._lib.ENGINE_get_default_RAND()
         assert e == backend._ffi.NULL
 
+    def test_osrandom_engine_implementation(self):
+        name = backend.osrandom_engine_implementation()
+        assert name
+        if sys.platform.startswith('linux'):
+            assert name in ['getrandom', '/dev/urandom']
+        elif sys.platform == 'win32':
```

Comment:
Our CI can't cover this branch (we don't get coverage from our windows builders, although this is being executed) so we'll need to figure out a way to rewrite this. Maybe make the asserts a separate function that takes a string `_engine_name_assert(sys.platform)` and then

```python
def _engine_name_assert(platform):
    if platform.startswith('linux'):
        assert name in ['getrandom', '/dev/urandom']
   elif platform == 'win32':
        assert name == 'CryptGenRandom'
```

and then a separate test for `_engine_name_assert` that explicitly passes `win32`?